src/zabapgit_syntax_highlighter.prog.abap

```diff
@@ -0,0 +1,762 @@
+            CONTINUE.
+          ENDIF.
+      ENDCASE.
+
+      lv_prev_token = <match>-token.
+      ASSIGN <match> TO <prev>.
+    ENDLOOP.
+
+    " Add entries refering to parts of text that should not be formatted
+    LOOP AT ct_matches ASSIGNING <match>.
+      IF <match>-offset > lv_last_pos.
+        lv_length = <match>-offset - lv_last_pos.
+        ls_match-token  = c_token-none.
+        ls_match-offset = lv_last_pos.
+        ls_match-length = lv_length.
+        INSERT ls_match INTO ct_matches INDEX sy-tabix.
+      ENDIF.
+      lv_last_pos = <match>-offset + <match>-length.
+    ENDLOOP.
+
+    " Add remainder of the string
+    IF lv_line_len > lv_last_pos.
+      lv_length = lv_line_len - lv_last_pos.
+      ls_match-token  = c_token-none.
+      ls_match-offset = lv_last_pos.
+      ls_match-length = lv_length.
+      APPEND ls_match TO ct_matches.
+    ENDIF.
+
+  ENDMETHOD.  " order_matches.
+
+  METHOD format_line.
+
+    DATA:
+          lv_chunk     TYPE string,
+          lv_css_class TYPE string.
+
+    FIELD-SYMBOLS:
+          <match> TYPE ty_match.
+
+    LOOP AT it_matches ASSIGNING <match>.
+      lv_chunk = substring( val = iv_line off = <match>-offset len = <match>-length ).
+
+      CASE <match>-token.
+        WHEN c_token-keyword.
+          lv_css_class = c_css-keyword.
+        WHEN c_token-comment.
+          lv_css_class = c_css-comment.
+        WHEN c_token-text.
+          lv_css_class = c_css-text.
+        WHEN c_token-none.
+          CLEAR: lv_css_class.
+      ENDCASE.
+
+      lv_chunk = me->apply_style( iv_line  = lv_chunk
+                                  iv_class = lv_css_class ).
+
+      rv_line = rv_line && lv_chunk.
+    ENDLOOP.
+
+  ENDMETHOD.                    "format_line
+
+  METHOD apply_style.
+
+    DATA lv_escaped TYPE string.
+
+    lv_escaped = escape( val = iv_line  format = cl_abap_format=>e_html_attr ).
+    IF iv_class IS NOT INITIAL.
+      rv_line = |<span class="{ iv_class }">{ lv_escaped }</span>|.
+    ELSE.
+      rv_line = lv_escaped.
+    ENDIF.
+
+  ENDMETHOD.                    "apply_style
+
+  METHOD process_line.
+
+    DATA: lt_matches TYPE ty_match_tt.
+
+    IF strlen( iv_line ) = 0.
+      RETURN.
+    ENDIF.
+
+    lt_matches = me->parse_line( iv_line ).
+
+    me->order_matches( EXPORTING iv_line    = iv_line
+                       CHANGING  ct_matches = lt_matches ).
+
+    rv_line = me->format_line( iv_line    = iv_line
+                               it_matches = lt_matches ).
+
+  ENDMETHOD.                    " process_line
+
+ENDCLASS.                       " lcl_code_highlighter IMPLEMENTATION
+
+
+*----------------------------------------------------------------------*
+*       CLASS ltcl_code_highlighter definition
+*----------------------------------------------------------------------*
+CLASS ltcl_code_highlighter DEFINITION FINAL
+  FOR TESTING RISK LEVEL HARMLESS DURATION SHORT
+  INHERITING FROM cl_aunit_assert.
```

Comment:
inheriting from cl_aunit_assert (see class documentatino) is deprecated, use cl_abap_unit_assert